src/DataTypes/statepair_list.rs

```diff
@@ -18,7 +18,7 @@ pub struct DepthFirstWaitingStateList {
 pub trait PassedStateListExt {
     fn put(&mut self, pair: StatePair);
     fn has(&self, pair: &StatePair) -> bool;
-    fn zones(&self, key: &(LocationID, LocationID)) -> Vec<Federation>;
+    fn zones(&self, key: &(LocationID, LocationID)) -> Vec<&Federation>;
```

Comment:
We cannot return a reference to a vec as not all the trait implementations use vectors, `PassedStateListVec` does but `PassedStateListFed` and `WaitingStateList` do not. We cannot construct a new vector and return a reference to it as it would go out of scope. I think the current solution is the most general

Reply:
Hmm good point in that case we could probably use a slice `&[Federation]` instead, although with VecDeque that will be funky if we push both front and back.

But we can just leave for now